### PR TITLE
BaSP-MR-124: Subscription move post put logic

### PR DESCRIPTION
**Pull request by Gonzarey96**
With the integration of redux in our app we can now effectively manage the state of our application, resulting in smoother data flow, easier tracking of changes, and improved performance.
In this pull request, I implement redux logic for adding or editing a subscription in this way:

Move the logic of the post, put it to the thunk file in the subscription folder, added the actions and reducer necessaries.
Replace the useState to get the data for useSelector

### CODE REVIEW

**Comment by Federico-Peralta** [comment]
[[Subscription] - Move Post and put logic](https://trello.com/c/9U2LtQd2/124-subscription-move-post-and-put-logic)

File: src/Components/Subscriptions/Form/index.jsx [hunk 1]
```diff
@@ -3,126 +3,80 @@ import styles from './form.module.css';
...
-        setAlertMessage('Subscription updated!');
-        setShowSuccessAlert(true);
-      } else if (!response.ok) {
-        setAlertMessage(data.message);
-        setShowSuccessAlert(true);
+      if (id) {
+        const data = await dispatch(updateSubscription(users, id));
+        showSuccesModal({ message: 'Success' });
+        setBodyModal(data.msg);
+        console.log(data);
```
Comment (fernandofernandezuy) [inline src/Components/Subscriptions/Form/index.jsx]: remove this console.log please

File: src/Redux/subscriptions/thunks.js [hunk 2]
```diff
@@ -2,62 +2,116 @@ import {
...
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(subscription)
       }
-    } catch (error) {
-      dispatch(deleteSubscriptionsError(error));
-      throw error;
+    );
+    const data = await response.json();
+    console.log(data);
```
Comment (fernandofernandezuy) [inline src/Redux/subscriptions/thunks.js]: same here

**Review by EmaBaldor — APPROVED** [review]
Good Job!

File: src/Components/Subscriptions/Form/index.jsx [hunk 3]
```diff
@@ -3,126 +3,80 @@ import styles from './form.module.css';
...
   const { id } = useParams();
   const history = useHistory();
-  const [classes, setClasses] = useState([]);
-  const [members, setMembers] = useState([]);
-  const [subscription, setsubscription] = useState([]);
+  const dispatch = useDispatch();
+  const classes = useSelector((state) => state.classes.data.data || []);
+  const members = useSelector((state) => state.members.data.data || []);
+  const subscription = useSelector((state) => state.subscriptions.data);
+
```
Comment (marcohzo) [inline src/Components/Subscriptions/Form/index.jsx:18]: delete this line

**Review by marcohzo — APPROVED** [review]

**Comment by LucreciaV** [comment]
Good job!

**Pull request merged**